### PR TITLE
Create pipeline-build-deploy-app.md

### LINKED ISSUE #403: Guide for source to deployed app using pipelines
### Add label "documentation-id"
- Add the `documentation-id` bug label to this issue.

### What version of Kabanero does this apply to?
- Latest (0.8.0 and above)

### Specify what you want documented (required):
Complete the following information:

#### To create a topic:
- Specify a link to the section in the navigation where the topic should go: Under pipelines guides
- Specify a first draft of the topic that you want added: Please check last comment in https://github.com/kabanero-io/kabanero-pipelines/issues/282

### PR DESCRIPTION
**Pull request by jantley-ibm**
closes https://github.com/kabanero-io/docs/issues/403

### CODE REVIEW

**Review by aadeshpa — APPROVED** [review]

**Review by aadeshpa — APPROVED** [review]
looks good

File: publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md
Comment (SueChaplain) [inline publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md]: For this guide, we will use ~~the~~ one of the default pipelines, `java-openliberty-build-deploy-pl`, ~~that~~ which  is activated ......

File: publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md
Comment (SueChaplain) [inline publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md]: - volumen / volume
- backtick `nfs-pv.yaml`

File: publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md
Comment (SueChaplain) [inline publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md]: Should really say "Change into the ...."

File: publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md
Comment (SueChaplain) [inline publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md]: I would make this sentence part of the last sentence rather than a bullet.

File: publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md
Comment (SueChaplain) [inline publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md]: .... follow the steps in the **Secrets section** of....

File: publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md
Comment (SueChaplain) [inline publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md]: .... follow the steps in the **Secrets section** of....

File: publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md
Comment (SueChaplain) [inline publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md]: I think I would lead into the list slightly differently as I was initially unsure what to do ... i.e. "Read the following information to determine ....."

File: publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md
Comment (SueChaplain) [inline publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md]: - Set up certificates or specify an insecure connection...? 
- see **Transport layer security (TLS) verification for image registry access in pipelines** in .....(makes it easier to read)

File: publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md
Comment (SueChaplain) [inline publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md]: .... dashboard with a webhook extension.... ?

File: publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md
Comment (SueChaplain) [inline publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md]: ... see **Checking the status of the pipeline run** in....

File: publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md
Comment (SueChaplain) [inline publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md:63]: The steps are butting right up to the subtitle here, which doesn't look great. I initially thought it was your curly braces, but it isn't.

BTW why do you not link like this rather than use curlies? Is that basic markdown or an extension they've added to the build. I didn't know you could do that! 

```
[Creating a secret and linking to a service account](#creating-a-secret-and-linking-to-a-service-account)
```

To add the space I just inserted `<br>` under the heading.

File: publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md
Comment (SueChaplain) [inline publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md]: The indentation goes wrong here, so step 2 says step 1.

You need to indent all your code blocks and text under each step to keep the flow. e.g. 

```
1. Find the service account of your application.


    oc get deployments -n kabanero

    NAME                                      READY   UP-TO-DATE   AVAILABLE   AGE
    codeready-operator                        1/1     1            1           28h
    java-openliberty-0-2-26                   1/1     1            1           25h
    kabanero-cli                              1/1     1            1           28h
    kabanero-landing                          1/1     1            1           28h
    kabanero-operator                         1/1     1            1           28h
    kabanero-operator-admission-webhook       1/1     1            1           28h
    kabanero-operator-collection-controller   1/1     1            1           28h
    kabanero-operator-stack-controller        1/1     1            1           28h
    

    Make a note of the name of the deployment, for example, `java-openliberty-0-2-26`.

2. Use the deployment name to find the `serviceaccount`.
    ```
    oc get deployments java-openliberty-0-2-26 -o yaml --output="jsonpath={.spec.template.spec.serviceAccount}"

    java-openliberty-0-2-26
    ```
    Note the `serviceaccount`; this is the `serviceaccount` to which the secret will be linked.
```

File: publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md
Comment (SueChaplain) [inline publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md]: ... other than the one found in... ?

Comment (SueChaplain) [reply publish/pipeline-build-deploy-app/pipeline-build-deploy-app.md]: Being picky, we should try and use present tense & active voice a bit more ... but it takes time to fix all these things in the input and I know we don't have much time. E.g 

The token will be a password for the creation of the secret.

The token is a password for creating the secret.

**Review by SueChaplain — APPROVED** [review]
LGTM. We should get this staged to double check the list items ... or I can test in another local build after merging.

**Comment by jantley-ibm** [comment]
@SueChaplain we have technical approval by @aadeshpa , could you go ahead and merge?

**Pull request merged**
